hypothesis/checkmate
# Use celery to update the blocklists and read from the db

For: https://github.com/hypothesis/checkmate/issues/37

This PR:

 * Makes the DB mandatory
 * Adds celery config and a worker
 * Removes the DB updates from the scripts
 * Adds a task for updating the DB instead
 * Duplicates some of the file parsing logic, but makes it separate from the existing system

This doesn't:

 * Actually do anything again!
 * Until there's a task runner queuing tasks nothing will happen
 * This doesn't remove the original mechanism, that's still what's making this work

# Review notes

This is all in heavy flux. We intend to remove the S3 file base system quickly for an admin interface. For that reason I've not really gone to down on testing this as it is right now (at least in their current form) as the following bits will be going away:

 * File downloading and parsing
 * The bulk update mechanism (as it'll be CRUD)
 * Both of these will return in some format though for URLHaus

# Testing notes

Script to trigger the task:
```python
from checkmate.async.tasks import sync_blocklist

sync_blocklist.delay()
```

* Run `make services`
* Run `CHECKMATE_BLOCKLIST_URL=<s3 URL> make dev`
* Run the above script
* Run `make sql`
* `select * from custom_rule;` - There should be rules in there now from the list
* Visit: http://0.0.0.0:9099/api/check?url=http://nautil.us
* You should see publisher blocked

Separately there's also a make dev command now so:

* Run `docker stop checkmate_postgres_1; docker rm checkmate_postgres_1; make services`
* Run `make devdata`
* Run `make sql`
* `select * from custom_rule;` - There should be rules in there now from `checkmate/scripts.py:update_dev_data()`

### Deployment notes

**The DB is now mandatory! We must add connection details for that.**

See: https://hypothes-is.slack.com/archives/CR3E3S7K8/p1607093496497300?thread_ts=1605260202.412800&cid=CR3E3S7K8

**Rabbit is also mandatory! We must add connection details for that.**

The settings are:

```
Prod
DATABASE_URL=postgres://checkmate:<PW_FROM_VAULT>@checkmate-prod-db.svc.hypothes.is/checkmate

QA
DATABASE_URL=postgres://checkmate:<PW_FROM_VAULT>@checkmate-qa-db.svc.hypothes.is/checkmate
```
